pkgs/build-support/dotnet/build-dotnet-module/default.nix

```diff
@@ -73,6 +71,7 @@
 , buildType ? "Release"
   # If set to true, builds the application as a self-contained - removing the runtime dependency on dotnet
 , selfContainedBuild ? false
+, useAppHost ? true
```

Comment:
Is this flag used anywhere? Is it useful for it to be exposed?

Reply:
I don't see any benefit for us to build .NET packages from source.

Are any other development environment's packages build by us? I don't see this with node ones at least (and Rust ones are always build from source but not directly by us).
What would the benefits of building it ourselves be, other than increasing maintenance cost even more?

Reply:
One of the great things about nixpkgs is being able to easily patch or reconfigure any package used on your system.

I just used newtonsoft-json as a test case because it's simple and ubiquitous.  I don't plan on proposing any major changes until it can be done with a reasonable maintenance burden.

Surely anything that can be built from source should be.  The fact that you don't see the benefit of this makes me worry that it's an uphill battle.  I'd love to discuss that but it's a little off-topic for here.

Reply:
> Surely anything that can be built from source should be. The fact that you don't see the benefit of this makes me worry that it's an uphill battle. I'd love to discuss that but it's a little off-topic for here.

If you want, you can open an issue and we can discuss that there

Reply:
If you can't disable this, when you try to build a project using netcoreapp2.0 (e.g. newtonsoft-json 11, which is still commonly used), you get something like:

```
/nix/store/1iwq17ry9s985r1bxyflcca51cpaipvq-dotnet-sdk-5.0.408/sdk/5.0.408/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(138,5): error NETSDK1068: The framework-dependent application host requires a target framework of at least 'netcoreapp2.1'. [/build/source/Src/Newtonsoft.Json.Tests/Newtonsoft.Json.Tests.csproj]
```

Reply:
I guess that's a good enough reason to leave it, we currently support building library projects and referencing them